packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Statements/ForStatement.cs

```diff
@@ -7,8 +7,19 @@
 
 namespace Microsoft.Generator.CSharp.Statements
 {
-    public sealed record ForStatement(AssignmentExpression? IndexerAssignment, ValueExpression? Condition, ValueExpression? IncrementExpression) : MethodBodyStatement, IEnumerable<MethodBodyStatement>
+    public sealed class ForStatement : MethodBodyStatement, IEnumerable<MethodBodyStatement>
     {
+        public AssignmentExpression? IndexerAssignment { get; set; }
+        public ValueExpression? Condition { get; set; }
+        public ValueExpression? IncrementExpression { get; set; }
+
+        public ForStatement(AssignmentExpression? indexerAssignment, ValueExpression? condition, ValueExpression? incrementExpression)
```

Comment:
i think from the syntax's perspective, this indexerAssignment does not have to be an `AssignmentExpression`?
the for statement is quite flexible and could do a lot of things.
For example, this is possible:
```
for (Console.WriteLine(i); i < 10; i++)
```
it might be meaningless but it is possible.
Considering we are kind of introducing a system of representing csharp syntax, we should not introduce too many constraints.

Reply:
For instance if we have this:
```csharp
	int i = 0;
	for (Console.WriteLine(i); i < 3; i++)
	{
		Console.WriteLine("wow");
	}
```
and it will output:
```
0
wow
wow
wow
```
just as expected.